src/types/stardust/block/payload/transaction.rs

```diff
@@ -253,6 +253,34 @@ mod rand {
                 },
             }
         }
+
+        /// Generates a random [`TransactionEssence`].
+        pub fn rand_spending(
```

Comment:
What's the purpose for this extra fn?

Reply:
`rand` possibly creates inputs and outputs that I don't want in my particular test-case scenario. I also need access to those inputs and outputs. For all created inputs `get_spending_transaction` has to return the same block. Since unspent outputs are not stored with the block, I need to separatedly insert them into the Outputcollection, and finally check that fetched block's payload (which is re-assembled through a `$lookup`) matches the one from the originally created random block.

Reply:
fixed